### PR TITLE
NGSTACK-851 back office skeleton

### PR DESCRIPTION
**Pull request by PetraMiseta**
NGSTACK-851 + NGSTACK-852

created skeleton for backoffice and integrated netgen conversations module 
(all css + js files for admin module taken from IOTA)

### CODE REVIEW

**Comment by iherak** [comment]
Could we please do a search&replace and replace everything `iota` related with something generic?

**Comment by emodric** [comment]
> Could we please do a search&replace and replace everything `iota` related with something generic?

Absolutely! No mention of iota should not remain.

File: assets/js/admin/components.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
+// This file contains only import, configuration and initialization of components
+//import addDocumentEventListeners from '../utilities/components/add-document-event-listeners';
```
Comment (pspanja) [inline assets/js/admin/components.js:2]: Can we rather keep this empty and add stuff as we need it?

File: assets/js/admin/toggle-edit-organization.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const editOrganizationBtn = document.querySelector('.js-edit-organization__btn');
```
Comment (pspanja) [inline assets/js/admin/toggle-edit-organization.js:1]: This seems project specific, can it be removed?

File: assets/sass/admin/_event_hosting.scss [hunk 3]
```diff
@@ -0,0 +1,376 @@
+.event-hosting-specialist,
```
Comment (pspanja) [inline assets/sass/admin/_event_hosting.scss:1]: We should remove this and other project specific stuff.

File: composer.lock
Comment (pspanja) [inline composer.lock:1]: Why was this removed (yarn.lock as well)?

File: templates/themes/app/backoffice/layout.html.twig [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    {% block meta %}{% endblock %}
+
+    {% for href in encore_entry_css_files('backoffice-index', 'app') %}
+        <link rel="stylesheet" type="text/css" href="{{ href }}">
+    {% endfor %}
+
+    {% block stylesheets %}{% endblock %}
+
+    <script src="{{ asset('bundles/netgenibexafieldtypehtmltext/build/app.js') }}"></script>
```
Comment (pspanja) [inline templates/themes/app/backoffice/layout.html.twig:21]: This should be removed, it's not used or installed.

File: composer.lock
Comment (emodric) [reply composer.lock:1]: Media site repo does not keep lock files. Only projects do.

File: assets/sass/backoffice/style.scss
Comment (pspanja) [inline assets/sass/backoffice/style.scss:1]: Let's remove stuff we don't currently use and are not likely to use soon: `filter_form`, `listing` and `modal`.

File: templates/themes/app/backoffice/layout.html.twig [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    <title>
+        {% block title %}
+            {{ 'layout.title'|trans }}
```
Comment (pspanja) [inline templates/themes/app/backoffice/layout.html.twig:25]: Missing translation.

File: templates/themes/app/backoffice/layout.html.twig [hunk 6]
```diff
@@ -0,0 +1,103 @@
+{% trans_default_domain 'admin' %}
```
Comment (pspanja) [inline templates/themes/app/backoffice/layout.html.twig:1]: Translation domain should be `backoffice`.

**Comment by pspanja** [comment]
@PetraMiseta please reply to each review comment with the commit that addresses it.

**Comment by pspanja** [comment]
@PetraMiseta @emodric @iherak rebased for added `peak_eng` siteaccess, moved routes to `config/app` and limited backoffice siteaccess to `peak_eng`, as per https://netgen.atlassian.net/browse/NGSTACK-867, in https://github.com/netgen/media-site/pull/182/commits/8935637985bc52b8953dbe1c6b84bf1fec33c3c1

**Comment by pspanja** [comment]
Rebased on `master` for Twig 3.9 support.

File: assets/sass/backoffice/style.scss
Comment (PetraMiseta) [reply assets/sass/backoffice/style.scss:1]: @pspanja done in https://github.com/netgen/media-site/commit/377146a0dee8bb41e39a983df420fe45aed2c94e

File: templates/themes/app/backoffice/layout.html.twig (hunk 4 above)
Comment (PetraMiseta) [reply templates/themes/app/backoffice/layout.html.twig]: @pspanja removed https://github.com/netgen/media-site/commit/2341a964c1414464f39cd7275a4fb744a3434507

File: assets/sass/admin/_event_hosting.scss (hunk 3 above)
Comment (PetraMiseta) [reply assets/sass/admin/_event_hosting.scss]: done here https://github.com/netgen/media-site/commit/c4c897f58797d6772823557b30c8fcde98037e10

File: assets/js/admin/toggle-edit-organization.js (hunk 2 above)
Comment (PetraMiseta) [reply assets/js/admin/toggle-edit-organization.js]: removed https://github.com/netgen/media-site/commit/c4c897f58797d6772823557b30c8fcde98037e10

File: assets/js/admin/components.js (hunk 1 above)
Comment (PetraMiseta) [reply assets/js/admin/components.js]: components updated here https://github.com/netgen/media-site/commit/c4c897f58797d6772823557b30c8fcde98037e10

**Comment by PetraMiseta** [comment]
Module is renamed to backoffice https://github.com/netgen/media-site/commit/c4c7339057cafddcf340673e1b6b48aabfbebbd4, IOTA specific code deleted and remained classes are renamed

File: assets/sass/backoffice/_icons.scss [hunk 7]
```diff
@@ -0,0 +1,159 @@
+%icon-dashboard {
+    background: url("data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTQgNEgxMFYxMkg0VjRaIiBzdHJva2U9IiMyNTJENDAiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIi8+CjxwYXRoIGQ9Ik00IDE2SDEwVjIwSDRWMTZaIiBzdHJva2U9IiMyNTJENDAiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIi8+CjxwYXRoIGQ9Ik0xNCAxMkgyMFYyMEgxNFYxMloiIHN0cm9rZT0iIzI1MkQ0MCIgc3Ryb2tlLXdpZHRoPSIyIiBzdHJva2UtbGluZWNhcD0icm91bmQiIHN0cm9rZS1saW5lam9pbj0icm91bmQiLz4KPHBhdGggZD0iTTE0IDRIMjBWOEgxNFY0WiIgc3Ryb2tlPSIjMjUyRDQwIiBzdHJva2Utd2lkdGg9IjIiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIvPgo8L3N2Zz4K") no-repeat center center;
+}
+
+%icon-my-country-profiles {
```
Comment (pspanja) [inline assets/sass/backoffice/_icons.scss:5]: This should be removed.

File: assets/sass/backoffice/_icons.scss [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+%icon-event-admin {
```
Comment (pspanja) [inline assets/sass/backoffice/_icons.scss:13]: Should be removed.

**Comment by pspanja** [comment]
@PetraMiseta we are missing menu item for the Dashboard page.

File: assets/sass/_variables.scss
[the remark this answers is not in the document]
Comment (TTrakostanec) [reply assets/sass/_variables.scss:1]: I think it would be better to separate

File: assets/sass/backoffice/_icons.scss (hunk 7 above)
Comment (PetraMiseta) [reply assets/sass/backoffice/_icons.scss]: done https://github.com/netgen/media-site/commit/f5531f7d7dea7ccb70e555fa113a0e299641e8fb

File: assets/sass/backoffice/_icons.scss (hunk 8 above)
Comment (PetraMiseta) [reply assets/sass/backoffice/_icons.scss]: done here https://github.com/netgen/media-site/commit/f5531f7d7dea7ccb70e555fa113a0e299641e8fb

**Comment by PetraMiseta** [comment]
> @PetraMiseta we are missing menu item for the Dashboard page.

added https://github.com/netgen/media-site/commit/3106fb06c86b0d9496b6c0ec1bde39f8ae5fa739

File: assets/sass/backoffice/_global_styles.scss [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+        a,
+        button {
+            margin-top: 1rem;
+        }
+    }
+}
+
+// Page actions styles (e.g. next page, discard, save)
+
+.ngsb-form-controls {
```
Comment (pspanja) [inline assets/sass/backoffice/_global_styles.scss:79]: Should be removed.

**Comment by pspanja** [comment]
With the extraction of backoffice SCSS variables, this looks good to me 👍

**Review by pspanja — DISMISSED** [review]

**Comment by pspanja** [comment]
@PetraMiseta there's a JS error in console coming from the Sidebar component (Firefox).

**Review by pspanja — APPROVED** [review]

**Review by emodric — APPROVED** [review]
